src/sagemaker/user_agent.py

```diff
@@ -25,9 +25,23 @@
 PYTHON_VERSION = '{}.{}.{}'.format(sys.version_info.major, sys.version_info.minor, sys.version_info.micro)
 
 
-def prepend_user_agent(client):
+def determine_prefix():
     prefix = 'AWS-SageMaker-Python-SDK/{} Python/{} {}/{} Boto3/{} Botocore/{}'\
         .format(SDK_VERSION, PYTHON_VERSION, OS_NAME, OS_VERSION, boto3.__version__, botocore.__version__)
+
+    try:
```

Comment:
would it be better to use `os.path.exists` rather than a try block?

Reply:
No, to do so wouldn't be pythonic and also exposes a race condition (the file could exist in line 1 but not exist or isn't accessible by the time line 2 is run).  Doing so via a try...except block is idomatic in Python.  This is called "EAFP" - see a good example here:

https://docs.python.org/3.4/library/os.html?highlight=eafp#os.access